fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt

```diff
@@ -0,0 +1,248 @@
+            lifecycleOwner = viewLifecycleOwner,
+            homeActivity = activity as HomeActivity,
+            navController = findNavController(),
+            menuButton = WeakReference(binding.menuButton),
+            hideOnboardingIfNeeded = interactor::onStartBrowsingClicked,
+        ).also { it.build() }
+
+        tabCounterView = TabCounterView(
+            context = requireContext(),
+            browsingModeManager = browsingModeManager,
+            navController = findNavController(),
+            tabCounter = binding.tabButton,
+        )
+
+        toolbarView?.build()
+
+        PrivateBrowsingButtonView(
+            button = binding.privateBrowsingButton,
+            browsingModeManager = browsingModeManager,
+            onClick = { mode ->
+                interactor.onPrivateModeButtonClicked(mode, userHasBeenOnboarded = false)
+            },
+        )
+
+        consumeFrom(store) {
+            tabCounterView?.update(it)
+        }
+
+        searchSelectorBinding.set(
+            feature = SearchSelectorBinding(
+                context = view.context,
+                binding = binding,
+                browserStore = store,
+                searchSelectorMenu = searchSelectorMenu,
+            ),
+            owner = viewLifecycleOwner,
+            view = binding.root,
+        )
+
+        searchSelectorMenuBinding.set(
+            feature = SearchSelectorMenuBinding(
+                context = view.context,
+                interactor = interactor,
+                searchSelectorMenu = searchSelectorMenu,
+                browserStore = store,
+            ),
+            owner = viewLifecycleOwner,
+            view = view,
+        )
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        if (browsingModeManager.mode == BrowsingMode.Private) {
+            activity?.window?.setBackgroundDrawableResource(R.drawable.private_home_background_gradient)
+        }
+
+        hideToolbar()
+    }
+
+    override fun onPause() {
+        super.onPause()
+
+        if (browsingModeManager.mode == BrowsingMode.Private) {
+            activity?.window?.setBackgroundDrawable(
+                ColorDrawable(
+                    ContextCompat.getColor(
+                        requireContext(),
+                        R.color.fx_mobile_private_layer_color_1,
+                    ),
+                ),
+            )
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+
+        onboardingView = null
+        homeMenuView = null
+        tabCounterView = null
+        toolbarView = null
+        _interactor = null
+        _binding = null
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+
+        homeMenuView?.dismissMenu()
+    }
+
+    private fun dispatchOnboardingStateChanges(state: OnboardingState) {
+        if (state != onboardingStore.state.onboardingState) {
```

Comment:
should this line of logic instead live where the `dispatchOnboardingStateChanges` lambda lives?

Reply:
I am not sure I followed this one? We don't have any visibility to the `onboardingStore.state` instead of the `OnboardingAccountObserver`.

Reply:
It's also a very awkward right now that we get the initial state from `OnboardingAccountObserver`. Will look into this as a followup.

Reply:
ahh gotcha gotcha. carry on.